dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx

```diff
@@ -0,0 +1,56 @@
+import { css } from '@emotion/react';
+import { palette, textSans, space } from '@guardian/source-foundations';
+import { SvgClock } from '@guardian/source-react-components';
+
+type Props = {
+	until?: number | undefined;
+};
+
+// work
+const deadlineStyles = css`
+	${textSans.xxsmall()};
+	color: ${palette.brand};
+	background-color: ${palette.brandAlt[400]};
+	right: 0;
+	display: flex;
+	align-items: center;
+	padding-right: ${space[1]}px;
+`;
+
+function getDaysBetween(first: Date, second: Date): number {
+	const ONE_DAY = 1000 * 3600 * 24;
+	return (second.getTime() - first.getTime()) / ONE_DAY;
+}
+
+export const getDeadlineText = (
+	date1: Date,
+	date2: Date,
+): string | undefined => {
+	const maxDays = 7;
+	const daysBetween = getDaysBetween(date1, date2);
+	if (daysBetween <= 0 || daysBetween > maxDays) return;
+	if (daysBetween <= 1) return 'Closing today';
+	if (Math.round(daysBetween) === 1) return 'Open for 1 more day';
+	return `Open for ${Math.round(daysBetween)} more days`;
+};
+
+function formatOptionalDate(date: number | undefined): Date | undefined {
+	if (date === undefined) return undefined;
+	const d = new Date(date * 1000);
+	return d;
+}
+
+export const Deadline = ({ until }: Props) => {
+	const untilDate = formatOptionalDate(until);
+	if (!untilDate) return null;
+	const now = new Date();
+	const deadlineText = getDeadlineText(now, untilDate);
+	if (!deadlineText) return null;
+
+	return (
+		<span css={deadlineStyles}>
+			<SvgClock size="xsmall" />
+			{deadlineText}
+		</span>
+	);
+};
```

Comment:
could we pull this into common rendering for both dcr and ar as they are the same, or are we moving away from common rendering completely?

Reply:
I think the latter - probably best to just put this in source. @abeddow91 may know more.